From: David Teigland <teigland@redhat.com>
Date: Mon, 10 Nov 2008 16:24:57 -0600
Subject: ocfs2: initialize stack_user lvbptr

The locking_state dump, ocfs2_dlm_seq_show, reads the lvb on locks where it
has not yet been initialized by a lock call.

Signed-off-by: David Teigland <teigland@redhat.com>
Acked-by: Joel Becker <joel.becker@oracle.com>
Signed-off-by: Mark Fasheh <mfasheh@suse.com>
---
 fs/ocfs2/stack_user.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

Index: linux-2.6.27-ocfs2/fs/ocfs2/stack_user.c
===================================================================
--- linux-2.6.27-ocfs2.orig/fs/ocfs2/stack_user.c
+++ linux-2.6.27-ocfs2/fs/ocfs2/stack_user.c
@@ -740,6 +740,9 @@ static int user_dlm_lock_status(union oc
 
 static void *user_dlm_lvb(union ocfs2_dlm_lksb *lksb)
 {
+	if (!lksb->lksb_fsdlm.sb_lvbptr)
+		lksb->lksb_fsdlm.sb_lvbptr = (char *)lksb +
+					     sizeof(struct dlm_lksb);
 	return (void *)(lksb->lksb_fsdlm.sb_lvbptr);
 }
 
